### ISSUE TITLE
Framework fails when running with password-protected keys

### ISSUE THREAD

**Issue by jmoreira-valory** [valory-xyz/open-autonomy#2088]
### Subject of the issue
When building and running a deployment with an ethereum keys.json file protected with a password, the following error is shown, and the launch is stopped:

Creating trader_abci_0 ... done

```
ERROR: for trader_tm_0  Container "f30163fec10d" is unhealthy.
Error occured bringing up the project
=================================================================================================================
Trying to get logs for trader_abci_0
Running the aea with aea, version 1.41.0
Checking keys
Checking to see if /agent_key/ethereum_private_key.txt exists
AEA key provided. Copying to agent.
ethereum key provided. Adding to agent.
Checking to see if /agent_key/cosmos_private_key.txt exists
AEA key not provided yet is required. Generating.
Generating key cosmos with a password!
cosmos key provided. Adding to agent.
Checking to see if /agent_key/solana_private_key.txt exists
Checking autonomy specific connection keys
Generating cosmos key for libp2p connection
[2023-10-19 12:50:03,359][ERROR] This is not a valid private key file: 'cosmos_private_key.txt'
 Exception: 'Error on key `cosmos_private_key.txt` load! Try to specify `password`: Error: KeyIsIncorrect('non-hexadecimal number found in fromhex() arg at position 0') '
Error: KeyIsIncorrect("Error on key `cosmos_private_key.txt` load! Try to specify `password`: Error: KeyIsIncorrect('non-hexadecimal number found in fromhex() arg at position 0') ")
error: This is not a valid private key file: 'cosmos_private_key.txt'
error:  Exception: 'Error on key `cosmos_private_key.txt` load! Try to specify `password`: Error: KeyIsIncorrect('non-hexadecimal number found in fromhex() arg at position 0') '

=================================================================================================================
Trying to get logs for trader_tm_0
Error: Encountered errors while bringing up the project.
```

### Your environment
- OS: Ubuntu
- Python version: 3.10.12
- Package Version: 0.13.1
- Anything else you consider helpful.

### Steps to reproduce
generate a keys.json file protected with a password
generate deployment providing password
run the deployment

**Comment by jmoreira-valory** [comment]
This issue is still not resolved in #2108 . Steps to reproduce:

Clone trader-quickstart at this PR https://github.com/valory-xyz/trader-quickstart/pull/55

Uncomment locally lines https://github.com/valory-xyz/trader-quickstart/pull/55/files#diff-91c4da23ec8a4f27b22bcaaf93a53e4a21cfeeb6a6977e9a3fdd994701ba3b40R641-R642

Run the script in a setting with a password-protected keyfiles. This is the output:

```
Detected an existing build. Using this one...
rm: cannot remove 'abci_build/nodes/node0/config/node_key.json': Permission denied
rm: cannot remove 'abci_build/nodes/node0/config/config.toml': Permission denied
rm: cannot remove 'abci_build/nodes/node0/config/genesis.json': Permission denied
rm: cannot remove 'abci_build/nodes/node0/config/priv_validator_key.json': Permission denied
rm: cannot remove 'abci_build/nodes/node0/data/priv_validator_state.json': Permission denied
You will need to provide sudo password in order for the script to delete part of the build artifacts.
Directory abci_build removed successfully.
Building deployment @ /home/jose/git/trader-quickstart/trader/trader_service/abci_build
WARNING: `dependencies` parameter not defined in the service
I[2023-11-17|17:56:21.859] Generated private validator                  module=main keyFile=node0/config/priv_validator_key.json stateFile=node0/data/priv_validator_state.json
I[2023-11-17|17:56:21.859] Generated node key                           module=main path=node0/config/node_key.json
I[2023-11-17|17:56:21.859] Generated genesis file                       module=main path=node0/config/genesis.json
Successfully initialized 1 node directories


Generated Deployment!


Type:                 docker-compose
Agents:               1
Build Length          10412



Running build @ /home/jose/git/trader-quickstart/trader/trader_service/abci_build
Creating trader_abci_0 ... done

ERROR: for trader_tm_0  Container "ed0ff6e7f294" is unhealthy.
Error occured bringing up the project
===================================================================================================================================================
Trying to get logs for trader_abci_0
Running the aea with aea, version 1.41.0.post1
Checking keys
Checking to see if /agent_key/ethereum_private_key.txt exists
AEA key provided. Copying to agent.
ethereum key provided. Adding to agent.
Checking to see if /agent_key/cosmos_private_key.txt exists
AEA key not provided yet is required. Generating.
Generating key cosmos with a password!
cosmos key provided. Adding to agent.
Checking to see if /agent_key/solana_private_key.txt exists
Checking autonomy specific connection keys
Generating cosmos key for libp2p connection
[2023-11-17 17:56:29,184][ERROR] This is not a valid private key file: 'cosmos_private_key.txt'
 Exception: 'Error on key `cosmos_private_key.txt` load! Try to specify `password`: Error: KeyIsIncorrect('non-hexadecimal number found in fromhex() arg at position 0') '
error: This is not a valid private key file: 'cosmos_private_key.txt'
error:  Exception: 'Error on key `cosmos_private_key.txt` load! Try to specify `password`: Error: KeyIsIncorrect('non-hexadecimal number found in fromhex() arg at position 0') '
Error: KeyIsIncorrect("Error on key `cosmos_private_key.txt` load! Try to specify `password`: Error: KeyIsIncorrect('non-hexadecimal number found in fromhex() arg at position 0') ")


```

**Comment by jmoreira-valory** [comment]
I wonder if the issue might be related to these 2 lines in trader-quickstart:

```
export OPEN_AUTONOMY_PRIVATE_KEY_PASSWORD="$password" && poetry cdrun autonomy deploy build "../../$keys_json_path" --n $n_agents -ltm
(...)
export OPEN_AUTONOMY_PRIVATE_KEY_PASSWORD="$password" && poetry run autonomy deploy run --build-dir "$directory" --detach
```

But I think not, because the deployment is built, and it seems the agent is correctly fetching ethereum keys.
